up.sh

```diff
@@ -40,48 +39,53 @@ function check_dependencies() {
     echo "The AWS_DEFAULT_REGION env variable must be set. Default is usually eu-west-2."
     exit 1
   fi
-  if [[ ! -z "${NODE_VERSION}" && ! -z "${NODE_BIN}" ]]; then
-    echo "Both NODE_VERSION and NODE_BIN cannot be set at the same time."
+  if [[ ! -z "${NODE_VERSION}" && ! -z "${NODE_BIN_PATH}" ]]; then
+    echo "Both NODE_VERSION and NODE_BIN_PATH cannot be set at the same time."
     echo "Please use one or the other."
     exit 1
   fi
 }
 
 function run_terraform_apply() {
-  local node_bin_path="${NODE_BIN}"
+  local node_version_arg="latest"
   if [[ ! -z "${NODE_VERSION}" ]]; then
-    (
-      cd /tmp
-      rm -rf sn_node-${NODE_VERSION}
-      mkdir sn_node-${NODE_VERSION}
-      aws s3 cp s3://sn-node/sn_node-${NODE_VERSION}-x86_64-unknown-linux-musl.tar.gz .
-      tar -C sn_node-${NODE_VERSION} -xvf sn_node-${NODE_VERSION}-x86_64-unknown-linux-musl.tar.gz
-    )
-    node_bin_path="/tmp/sn_node-${NODE_VERSION}/sn_node"
+      node_version_arg="${NODE_VERSION}"
+  elif [[ ! -z "${NODE_BIN_PATH}" ]]; then
+    if [[ -d "${NODE_BIN_PATH}" ]]; then
+      echo "The node bin path must be a file"
+      exit 1
+    fi
+    local path=$(dirname "${NODE_BIN_PATH}")
+    node_version_arg="$testnet_channel"
+    archive_path="/tmp/sn_node-${node_version_arg}-x86_64-unknown-linux-musl.tar.gz"
```

Comment:
the build / bin isnt necessarily musl.

heaptrack needs it to _not_ be musl (but, if we pull in a musl build, heaptrack just wont run... which is fine for those situations where we dont want it)

Reply:
Ah, this is just a name for the archive, it doesn't actually affect the contents; however, I can see why it would be confusing.

I'll refactor this so it actually just passes the S3 URL to Terraform, and then a different URL can be used.

Reply:
OK, I've now modified the commit for this. The term 'custom' will get used rather than 'musl'.